### ISSUE TITLE
[BUG] "+" character removed while searching

### ISSUE THREAD

**Issue by NitroSniper** [excalith/excalith-start-page#68]
### Describe The Bug
When searching with the + character anywhere in the buffer. It will be removed when searching which can change the normal results. Even force search will have this issue. 

### To Reproduce

1. Go to [StartPage](https://excalith-start-page.vercel.app/)
2. Search either:
* "How to do string split in c++"  => How to do string split in c
* "(3 + 4) + (4 + 3)"  => "(3   4)   (4   3)"
3. You'll see current search result being changed

### Expected Behavior
For the search result to not be changed

### Specs

- OS: Window 10 & Arch Linux
- Browser: Firefox 121

**Comment by excalith** [comment]
Hey @NitroSniper!

Nice catch, I need to use [encodeURIComponent](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) to properly encode the string in order to work with requests.

The string should be encoded to ouput something like this in order for engines to properly handle symbols:
```
https://www.google.com/search?q=How%20to%20do%20string%20split%20in%20c%2B%2B
```

Will start working on it as soon as possible and release as a hotfix when it is ready.

Cheers!

**Comment by excalith** [comment]
Fixed with PR #69, releasing the hotfix as `3.1.2` now
